Source/Rhetos.Utilities/ApplicationConfiguration/ConfigurationSources/JsonSource.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Rhetos.Utilities.ApplicationConfiguration.ConfigurationSources
+{
+    public class JsonSource : IConfigurationSource
+    {
+        private readonly string _jsonText;
+
+        public JsonSource(string jsonText)
+        {
+            _jsonText = jsonText;
+        }
+
+        public IDictionary<string, object> Load()
+        {
+            var parsedJson = JObject.Parse(_jsonText);
+            return GetKeysFromObject(parsedJson, "");
+        }
+
+        private static readonly JTokenType[] _allowedJTokenTypes = { JTokenType.Boolean, JTokenType.String, JTokenType.Integer, JTokenType.Float, JTokenType.Object };

```

Comment:
I don't fully understand _allowedJTokenTypes validation? What would happen in some of the other token types were accepted by this method, for example Date, Time, Guid and Null values? What should developer/admin do if the "Json token type" exception occurs?

Reply:
Discussed and kept as is. Added fix for problems with implicit DateTime parsing.